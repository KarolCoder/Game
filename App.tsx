import {MainNavigator} from '@/navigation/MainNavigator';
import React from 'react';
import {useTheme} from 'react-native-paper';
import styled from 'styled-components/native';

const App = () => {
  const {colors} = useTheme();
  return (
    <BlackView backgroundColor={colors.background}>
      <SafeArea>
        <MainNavigator />
      </SafeArea>
    </BlackView>
  );
};

const SafeArea = styled.SafeAreaView`
  flex: 1;
`;
const BlackView = styled.View<{backgroundColor: string}>`
  flex: 1;
  background-color: ${({backgroundColor}) => backgroundColor};
`;

export default App;
