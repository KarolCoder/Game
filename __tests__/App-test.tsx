/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';

it('renders correctly', async () => {
  const result = renderer.create(<App />);
  await act(async () => {
    expect(result).toMatchSnapshot();
    return;
  });
});
