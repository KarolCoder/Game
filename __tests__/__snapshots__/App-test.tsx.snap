// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  backgroundColor="#f6f6f6"
  style={
    Array [
      Object {
        "backgroundColor": "#f6f6f6",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenStack
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          forwardedRef={[Function]}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          stackPresentation="push"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backgroundColor="rgb(255, 255, 255)"
            color="rgb(0, 122, 255)"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={true}
            hideBackButton={false}
            largeTitleHideShadow={false}
            title="Welcome"
            titleColor="rgb(28, 28, 30)"
            topInsetEnabled={false}
            translucent={false}
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/assets/images/starWars.jpeg",
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": "100%",
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 0.87)",
                        "fontFamily": "System",
                        "fontWeight": "500",
                        "writingDirection": "ltr",
                      },
                      Array [
                        Object {
                          "fontSize": 20,
                          "letterSpacing": 0.15,
                          "lineHeight": 30,
                          "marginVertical": 2,
                        },
                        Array [
                          Object {
                            "color": "#6200ee",
                            "marginTop": 40,
                          },
                        ],
                      ],
                    ],
                  ]
                }
                textColor="#6200ee"
              >
                Star Wars Game
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "marginTop": "auto",
                      "paddingBottom": 0,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "paddingTop": 0,
                      "width": "100%",
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "#6200ee",
                      "borderColor": "transparent",
                      "borderRadius": 4,
                      "borderStyle": "solid",
                      "borderWidth": 0,
                      "elevation": 2,
                      "marginBottom": 16,
                      "minWidth": 64,
                      "shadowColor": "#000000",
                      "shadowOffset": Object {
                        "height": 0.75,
                        "width": 0,
                      },
                      "shadowOpacity": 0.24,
                      "shadowRadius": 1.5,
                    }
                  }
                >
                  <View
                    accessibilityHint="bottom button"
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        Object {
                          "overflow": "hidden",
                        },
                        Object {
                          "borderRadius": 4,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "letterSpacing": 1,
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              undefined,
                              Object {
                                "textTransform": "uppercase",
                              },
                              Object {
                                "color": "#ffffff",
                                "fontFamily": "System",
                                "fontWeight": "500",
                              },
                              Object {
                                "fontFamily": "System",
                                "fontWeight": "500",
                              },
                              undefined,
                            ],
                          ]
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000000",
                                "fontFamily": "System",
                                "fontWeight": "400",
                              },
                              Object {
                                "textAlign": "left",
                              },
                              Array [
                                Object {
                                  "textAlign": "left",
                                },
                                Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "System",
                                  "fontWeight": "500",
                                  "writingDirection": "ltr",
                                },
                                Array [
                                  Object {
                                    "fontSize": 20,
                                    "letterSpacing": 0.15,
                                    "lineHeight": 30,
                                    "marginVertical": 2,
                                  },
                                  undefined,
                                ],
                              ],
                            ]
                          }
                        >
                          Play
                        </Text>
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenStack>
    </RNCSafeAreaProvider>
  </RCTSafeAreaView>
</View>
`;
